docs/kepler_model_server/pipeline.md

```diff
@@ -66,34 +44,32 @@ Full|WORKLOAD_FEATURES, SYSTEM_FEATURES|All
 
 > node information refers to value from [kepler_node_info](../design/metrics.md#kepler-metrics-for-node-information) metric.
 
-## Learning methods
-
-`trainer` is an abstract to define the learning method applies to each feature group with each given power labeling source. `Trainer` class has 9 abstract methods.
-
-1. load previous checkpoint model via implemented `(i) load_local_checkpoint` or `(ii) load_remote_checkpoint`. If the checkpoint cannot be loaded, initialize the model by calling implemented `(iii) init_model`.
+## Power isolation
 
-2. load and apply scaler to input data
+`isolate` is a mechanism to separate the power portion that is consumed on the node in an idle state or the power portion that is consumed by operating systems and control plane processes. These portions of power is more than zero even if the metric utilization of workload is zero. We called the models those are trained after isolating these power portions as `DynPower` models. At the same time, the models those are trained without isolation are called `AbsPower` models. `DynPower` model is used to estimate `container power` and `process power` while `AbsPower` model is used to estimate `node power`.
 
-3. call implemented `(iv) train` and save the checkpoint via `(v) save_checkpoint`
+There are two common available `isolators`: *ProfileIsolator* and *MinIdleIsolator*. 
 
-4. check whether to archive the model and push to database via `(vi) should_archive`. If yes, 
-      
-      4.1.  get trainer-specific basic metadata via `(vii) get_basic_metadata`
+*ProfileIsolator* relies on profiled background powers (profiles) and removes resource usages by system processes from the training while *MinIdleIsolator* assumes minimum power as an idle power and includes resource usages by system processes in the training. 
 
-      4.2. fill with required metadata, save it as metadata file (metadata.json)
+The pipeline with *ProfileIsolator* will be applied first if the profile that matches the training `node_type` is available. Otherwise, the other pipeline will be applied. 
```

Comment:
If the `profile data` that matches a given `node_type` exist, the pipeline will use the *ProfileIsolator* to preprocess the training data. Otherwise, the the pipeline will applied another isolation mechanism, such as the *MinIdleIsolator*.